### ISSUE TITLE
Column Lineage Handling For Nested Token and Inline Comment

### ISSUE THREAD

**Issue by eeroel** [reata/sqllineage#172]
Hello,

First of all thank you @reata for this cool library! I was experimenting with a large set of SQL scripts and found a few cases for which the column extraction fails. The failures are mainly due to two issues, as far as I can see:
1. improper handling of nested tokens when extracting column names and
2. statements with window functions not handled properly

I'll paste the test cases here:
```
def test_comment_after_column():
    sql = """INSERT OVERWRITE TABLE tab1
    SELECT
    a.col1
    --, a.col2
    , a.col3
    FROM
    tab2 a
    """
    assert_column_lineage_equal(
        sql, [("tab2.col1", "tab1.col1"), ("tab2.col3", "tab1.col3")]
    )


def test_cast_with_comparison():
    sql = """
    INSERT OVERWRITE TABLE tab1
    select cast(col1=1 as int) col1, col2=col3 col2
    from tab2;
    """
    assert_column_lineage_equal(
        sql, [("tab2.col1", "tab1.col1"), ("tab2.col2", "tab1.col2"), ("tab2.col3", "tab1.col2")]
    )


def test_complex_window_function():
    sql = """
    INSERT OVERWRITE TABLE tab1
    SELECT tab2.col1 AS col1
        , MAX(tab2.col1) OVER (PARTITION BY tab2.col2) AS col2
    FROM tab2
    """
    assert_column_lineage_equal(
        sql, [("tab2.col1", "tab1.col1"), ("tab2.col1", "tab1.col2"), ("tab2.col2", "tab1.col2")]
    )

def test_window_function_in_subquery():
    sql = """
    insert into tab1
    select rn from (
        select
            ROW_NUMBER() OVER (PARTITION BY col1, col2) rn
        from tab2
    ) sub
    where rn = 1
    ;
    """
    assert_column_lineage_equal(
        sql, [("tab2.col1", "tab1.rn"), ("tab2.col2", "tab1.rn")]
    )
```

I have ideas how to fix some of these so I'll open a PR to demonstrate.

**Comment by reata** [comment]
Thank you so much for submitting this issue as well as PR. I was also aware of both case, just hasn't decided which way to go yet.

1. Nested tokens
I observe column lineage parsing failure for complicated CASE clause. Token after keyword "WHEN" and keyword "THEN" can be a lot of things while I only covered the most simple case right now. 
For a solution, initially I was thinking of breaking _extract_source_raw_names into multiple small functions. But as in your PR name (haven't read the code yet), recursive calling might also be a good idea. Let see if this works.
And do let me know if you found any failure case of nested token other than CASE clause.

2. Window function
This is the part puzzles me most. "ROW_NUMBER() OVER (PARTITION BY col1, col2) rn" will be parsed as two separate identifier by sqlparse. So basically what I was doing is ignore "ROW_NUMBER()" as a simple function without parameter and handle anything after keyword "OVER". 
But this simply doesn't work for:  1) window function with parameter like "MAX(col) OVER ...".  2) simple function LIKE "SELECT now() AS col1", where we will ignore the now() -> col1 lineage for now.
Previously I was thinking maybe I need an exhaustive list of all window function for special handling in the end. But again, I haven't decided yet. I'll see your solution in the PR but it will be great help if you can also give your thought and implementation details here.

**Comment by reata** [comment]
Limiting the scope of this issue to nested token and inline comment. Closed via #173
